### ISSUE TITLE
 Game slow down (occasional crash) on startup with multiple faction mods; conflict with MRHVehicleSpawner

### ISSUE THREAD

**Issue by Eutyches** [MisterHLunaticwraith/MRHMilsimTools#26]
@MisterHLunaticwraith , thank you for your response here in GitHub to my original issue report and for your thoughtful messages in the Bohemia Interactive Forums. On behalf of the entire Praetorian Security Operations team, I thank you for your excellent work, for your open licensing (note: we conspicuously give credit to you and the MRHMilsimTools team basically everywhere possible), and for your laudable responsiveness as a developer.

Here is the (far) more specific issue report that I promised you -- i.e. the report that I should have submitted originally. I closed the last issue entirely, since I do not want anyone to stumble across it while reading about your mod and think that something is fundamentally broken. As a whole, MRHMilsimTools is stable. I hope that this new report will help us work together to solve the problem.

---

Arma 3 Version: 1.94 (stable)
CBA Version: 3.12.1 (stable)
ACE3 Version: 3.12.6 (stable)
Milsim Tools Version: 1.17.3 (stable)

Mods: A large additional mod pack **with many factions** (the number of factions seems to be what matters) 
Specifically, we tested with all RHS, Project OPFOR, & BAF mods, plus a few custom unit-specific Alive-based factions.

Description: At game startup, as various.rpt logs from players who have had this problem show, MRHVehicleSpawner hangs while going through its faction checklist, causing a 1-3 minute delay (frozen menu before being able to play as normal) for a minority of players and a possible game crash for a small minority.

Steps to reproduce: Launch Arma 3 with MRHMilsimTools and a large number of mods that create a long list of custom factions.

[Sample .rpt log (pastebin)](https://pastebin.com/bWFjkyy6)
Note: This particular player has other Arma 3 stability-related errors (DX11, memory, etc.). Ignore those and focus on the "MRH spawner" checklist, which is near the end of the log.

Temporary workaround that helped with diagnosis: Remove the addon MRHVehicleSpawner entirely. No slow down on start. The rest of MRHMilsimTools works as intended. We found this workaround by trial and error. Players who experience the hang at start with the full MRHMilsimTools mod no longer experience any slow down whatsoever once MRHVehicleSpawner has been removed.

**Comment by MisterHLunaticwraith** [comment]
Yes the vehicle spawner reads all the vehicle configs to compile the main list. It's a consuming process. It's an old addon I've been meaning to update. Best I can do is leave the processing to the server (would be best practice should have done that in the 1st place). It will run only once at mission start and on the server. But I must say that my unit and I use as many factions mods as you and don't get the frozen menu issue. We do not use BAF mods though. I will fix the problem in the next update

**Comment by MisterHLunaticwraith** [comment]
Thanks !After browsing the spawner code I think it's old and needs a complete revamp (it was one of the first addon I made for this mod. So hold your horses because the next update  might take some time to see the light of day.

**Comment by MisterHLunaticwraith** [comment]
@Eutyches I've strayed to other functionalities so the full makeover is not for any time soon however I've passed the reading of config to the server only. This should fix the issues you were having. Are you still experiencing it? Is it okay to close the issue?
